### PR TITLE
Message type for candidate delta

### LINKED ISSUE #32: as a network user I want to be able to broadcast candidate deltas
this will allow for voting for the most popular candidate
a candidate delta contains
- a hash of the content of the delta
- the producer's PeerIdentifier
- the hash of the delta preceding the candidate

### PR DESCRIPTION
**Pull request by monsieurleberre**
closes #32

### CODE REVIEW

File: src/Delta.proto [hunk 1]
```diff
@@ -37,12 +39,26 @@ package Catalyst.Protocol.Delta;
...
+ *
+ * DeltaHash: The hash computed for the current delta produced by ProducerId.
+ *	This is meant to be used when voting for most popular hashes.
+ * PreviousDeltaDfsHash: DFDS address for the content of the delta preceding this candidate.
+ * ProducerId: Identifier of the producer of the candidate.
+ */
+message CandidateDelta {
+	bytes Hash = 1;
+	Common.PeerId ProducerId = 2;
+	bytes PreviousDeltaDfsHash = 3;
```
Comment (PaulineBernat) [inline src/Delta.proto:63]: Am I correct to assume that you will need this in order to compare CandidateDelta.Hash with Delta locally computed when implementing sharding? Is it useful at the moment? If no secondary chain, every node should have the same PreviousDeltaDfsHash right?

**Review by PaulineBernat — APPROVED** [review]
Looks good to me, thank you

**Review by franssl — APPROVED** [review]
All looks good to me.

File: src/Delta.proto (hunk 1 above)
Comment (nshCore) [reply src/Delta.proto:63]: I think it's to chain the deltas together.

**Review by nshCore — APPROVED** [review]

**Pull request merged**